content/getting-work-done-in-open-source.rst

```diff
@@ -0,0 +1,157 @@
+:title: How to get work done in open source
+:slug: getting-work-done-in-open-source
+:sortorder: 40
+:date: 2019-07-26 20:00
+
+After working in open source for a while, I've been on both sides of the code
+submission dance: proposing a change, and reviewing a change.
+
+Proposing a change to a project can feel harrowing: you don't know how the
+maintainers are going to respond to it, you don't know whether you've followed
+all the unwritten rules for that project. You may submit a change that no one
+reviews for weeks or months, or you may be immediately charged with explaining
+yourself and making completely transformative changes until your original change
+isn't recognizable as your own any more.
+
+But being a reviewer is not an easy task either. You have to try to put yourself
+in the shoes of the submitter, who you most likely have never met, whose skill
+level and experience with the code base you cannot judge, and try to understand
+their problem and see the solution from their eyes. If the change is
+particularly hairy and the submitter has not explained it very well, it's
+tempting to gloss over the change and pick on the trivial things, or worse to
+just ignore the change.
+
+While the project maintainers should carry most of the responsibility for
+properly guiding a new contributor through the project's contribution processes
+and helping them to meet the maintainers expectations, there is a foolproof way
+that code submitters, regardless of the project, can speed the process along:
+
+**Write better commit messages**
+
+The most important part of a code change is not the code itself but the commit
+message. The prose you use to prepare the reviewer for the change is your best
+opportunity to get them on your side. Perfect spelling and conformity to style
+conventions are not what is important here (but I do still recommend you read up
+on the `guidelines`_). What matters is getting the discussion off on the right
+foot. The following is what has worked for me as a code contributor to get my
```

Comment:
As you already know I'm 100000% on board with these sentiments :-)  I find that by far the most common mistakes of submitters are either a) assuming the reviewers are all-seeing, all-knowing gurus who can instantly, telepathically understand the context behind the commit and the rationale for making the change this way, or worse, b) not even putting themselves in the shoes of the reviewer and considering how easy or hard the commit might be to review.  I'm not sure which is more common but the net effect is the same.  I wonder if it would be possible to squeeze in a sentence here calling that out more explicitly, without sounding as bolshy as I just did?  I very much like that above you said "You have to try to put yourself in the shoes of the submitter" but I think it applies equally the other way round too.

Reply:
Thanks for the review! I've added a sentence above that I hope captures this, what do you think?

Reply:
Looks great to me!